aten/src/ATen/core/intrusive_ptr.h

```diff
@@ -276,11 +275,37 @@ class intrusive_ptr final {
     return use_count() == 1;
   }
 
+  /**
+   * Returns an owning (!) pointer to the underlying object and makes the
+   * intrusive_ptr instance invalid. That means the refcount is not decreased.
+   * You *must* put the returned pointer back into a intrusive_ptr using
+   * intrusive_ptr::reclaim(ptr) to properly destruct it.
```

Comment:
It might be helpful to have an API to directly decrement the refcount of a raw pointer to an `intrusive_pointer_target`, instead of making people first make an intrusive pointer, and then move it out of scope.

Reply:
Not sure I like that. Basically, I see release/reclaim as a hack to temporarily pass a pointer through a C API. It should live as a raw pointer as briefly as possible and always go back into an intrusive_ptr because only intrusive_ptr knows how to handle the invariants.

Offering a second way to destruct it adds more code that needs to be aware of the invariants and make sure it knows to destruct at refcount zero and so on.

Reply:
I mean, sure, internally it can be implemented by constructing an intrusive pointer. All I'm saying is that an API of "if you release the pointer, you must call free on it" is simpler than "if you release the pointer, you must rewrap it in intrusive ptr exactly once".